### ISSUE TITLE
Hydra does not bind keys when using Japanese or Korean input modes

### ISSUE THREAD

**Issue by iferguson90** [sdegutis/mjolnir#124]
Hello! Hydra does not seem to bind hotkeys if it is launched while a Japanese or Korean input method is active.

For example, if I am currently using _Kotoeri_ (Japanese), and start Hydra with the sample config, I cannot open a REPL with `Ctrl-Alt-Cmd-R`. Even if I switch back to the U.S. input method and reload the config, the hotkeys will still not work unless I quit and reopen Hydra.

Likewise, if I start Hydra while the U.S. input method is active, the hotkeys will work, even if I then switch to a Japanese or Korean input method and/or reload the config.

The only input methods I have tested this with are _U.S. English_, _2-Set Korean_, and _Kotoeri_.

System details: MacBookPro10,1 (2012 Retina) with a US Keyboard, OS X 10.9.4, Hydra 1.0 b6. My system language is set to English.

To enable Japanese or Korean input: System Preferences > Keyboard > Input Sources. Check **[X] Show Input menu in menu bar**. Click the **[+]**, then select _Japanese_ or _Korean_ and then the appropriate input method. The Input menu usually shows up near the clock. A good mode to test with is the _Romaji_ input mode, which is part of the _Kotoeri_ (Japanese) input method. It should be mostly equivalent to typing with the U.S. input method (assuming that is your default layout), although Hydra will not bind hotkeys when it is launched and _Romaji_ is active.

Please let me know if you need clarification or additional testing for this bug.

**Comment by sdegutis** [comment]
@iferguson90 excellent bug report, thank you! I'm putting this in the 1.1 milestone for now.

**Comment by sdegutis** [comment]
Figured out why this probably is. The file `hotkey_translator.m` is probably getting a NULL `layoutData` variable in this case. Since this is run only once when Hydra is launched, a relaunch is the only way to rerun it. I should probably look for a "input method switched" event and re-run this code when that happens or something.

**Comment by sdegutis** [comment]
fwiw, the `next` branch has an updated version of that file, so nobody should modify this file on `master`.

**Comment by sdegutis** [comment]
There's two parts to this:
1. `hotkey.keycodes` † is not reloaded when you change input modes
2. For some input modes, there are no corresponding keycodes for certain ASCII characters

The first one is easy to do manually, but I don't know if there's an event called "the input mode changed" that we can hook into to automatically reload this table.

I'm not sure how to fix the second one. You can [see here](https://github.com/sdegutis/hydra/blob/9deb3a78832cc1f1742fcf4bea58604de48625f9/Hydra/ThirdParty/hotkey_translator.m#L33-L55) that we're literally just skipping a bunch of ASCII codes if they're not supported by this layout. Should we instead just map them directly to what they typically are, i.e. map `kVK_ANSI_A` to `"a"` etc? Not being a user of different input modes myself, I don't know if that would be appropriate. @tsl0922 and @iferguson90, thoughts?

† the table `hotkey.keycodes` is currently on the `next` branch and will be merged into `master` soon.

**Comment by sdegutis** [comment]
This is done on master.
